front/pages/api/data_sources/[user]/managed.ts

```diff
@@ -0,0 +1,192 @@
+    res.status(authRes.error.status_code).end();
+    return;
+  }
+  const auth = authRes.value;
+
+  if (!dataSourceUser) {
+    res.status(404).end();
+    return;
+  }
+
+  const dataSourceUserId = dataSourceUser.id;
+
+  const role = await auth.roleFor(dataSourceUser);
+
+  const where =
+    role === Role.ReadOnly
+      ? {
+          userId: dataSourceUser.id,
+          // Do not include 'unlisted' here.
+          visibility: "public",
+        }
+      : {
+          userId: dataSourceUser.id,
+          visibility: {
+            [Op.or]: ["public", "private"],
+          },
+        };
+  const dataSources = await DataSource.findAll({
+    where,
+    order: [["updatedAt", "DESC"]],
+  });
+
+  switch (req.method) {
+    case "POST": {
+      if (role !== Role.Owner) {
+        res.status(401).end();
+        return;
+      }
+      const providerId = "openai";
+      const maxChunkSize = 1024;
+      const dataSourceId = `Slack-Managed`;
+      const description = "Slack data source managed by Dust";
+      const modelId = "text-embedding-ada-002";
+
+      if (!req.body || !(typeof req.body.nango_connection_id == "string")) {
+        res.status(400).end();
+        return;
+      }
+
+      let nangoConnectionId = req.body.nango_connection_id as string;
+
+      // Enforce FreePlan limit: 1 DataSource.
+      if (dataSources.length >= 1 && auth.user().username !== "spolu") {
+        res.status(400).end();
+        return;
+      }
+
+      const dustProject = await DustAPI.createProject();
+      console.log("Created project: ", dustProject);
+      if (dustProject.isErr()) {
+        res.status(500).end();
+        return;
+      }
+
+      // The OpenAI credentials will be managed by Dust too.
+      let [providers] = await Promise.all([
+        Provider.findAll({
+          where: {
+            userId: auth.user().id,
+          },
+        }),
+      ]);
+      let credentials = credentialsFromProviders(providers);
+
+      const dustDataSource = await DustAPI.createDataSource(
+        dustProject.value.project.project_id.toString(),
+        {
+          dataSourceId: dataSourceId,
+          config: {
+            provider_id: providerId,
+            model_id: modelId,
+            splitter_id: "base_v0",
+            max_chunk_size: maxChunkSize,
+            use_cache: false,
+          },
+          credentials,
+        }
+      );
+      console.log("Created data source: ", dustDataSource);
+
+      if (dustDataSource.isErr()) {
+        res.status(500).end();
+        return;
+      }
+
+      const createdDataSource = await DataSource.create(
+        {
+          name: dataSourceId,
+          description: description,
+          visibility: "private",
+          config: JSON.stringify(dustDataSource.value.data_source.config),
+          dustAPIProjectId: dustProject.value.project.project_id.toString(),
+          userId: dataSourceUserId,
+        }
+      );
+
+      await Connector.create(
```

Comment:
How is the connectorId set on the DataSource object?

Reply:
It's not. We only set `connector.dataSourceId`. I have nothing against setting `dataSource.connectorId`, I just could not find how to do it with Sequelize :/